src/test/java/org/jenkinsci/plugins/docker/workflow/DockerDSLTest.java

```diff
@@ -149,7 +149,7 @@ private static void grep(File dir, String text, String prefix, Set<String> match
                 assumeDocker();
                 WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
                 p.setDefinition(new CpsFlowDefinition(
-                        "docker.image('maven:latest').inside {\n" +
+                        "docker.image('maven:latest').overrideEntrypoint {\n" +
```

Comment:
IMO, it seems like we are creating more problems for ourselves than we need to.

Correct me if I'm wrong but the previous behavior of `docker.image('foo').inside('--entrypoint cat'){...}` is equivalent to what is used as the default now.  Why can't we allow the user to add the command line argument where appropriate or omit it if it works without it?  At least this would provide users with a choice of how the container is used. Today there is no way for the end user to not use '--entrypoint'. This would work fine for `pipeline-model-definition` agent as well because we can pass in command args.